src/Main.purs

```diff
@@ -198,31 +199,36 @@ main = void do
   insertFilenames = foldr \x s -> maybe s (flip Set.insert s) x.filename
   loadNothing _ _ = pure Nothing
 
-  spawn' cmd args onExit = do
-    child <- Child.spawn cmd args Child.defaultSpawnOptions { stdio = Child.pipe }
-    outBuffer <- Ref.new ""
-    errBuffer <- Ref.new ""
-    fillBuffer (Child.stdout child) outBuffer
-    fillBuffer (Child.stderr child) errBuffer
+  spawn' outputStream cmd args onExit = do
+    let stdio =
+          case outputStream of
+            Stdout -> [ Just Child.Pipe, Just Child.Pipe, unsafePartial (Array.unsafeIndex Child.inherit 2) ]
+            Stderr -> [ Just Child.Pipe, unsafePartial (Array.unsafeIndex Child.inherit 1), Just Child.Pipe ]
+
+    child <- Child.spawn cmd args Child.defaultSpawnOptions { stdio = stdio }
+    buffer <- Ref.new ""
+    fillBuffer buffer $
+      case outputStream of
+        Stdout -> Child.stdout child
+        Stderr -> Child.stderr child
     Child.onExit child \status ->
       case status of
         Child.Normally n -> do
-          stdout <- Ref.read outBuffer
-          stderr <- Ref.read errBuffer
-          onExit { stdout, stderr, exitCode: n }
+          output <- Ref.read buffer
+          onExit { output, exitCode: n }
```

Comment:
Was the other output channel just being completely ignored? Is there any reason why it might be worth capturing both?

Reply:
Yeah, we weren’t doing anything with the stream that we determined didn’t contain the json errors. I can’t think of a reason to capture both; the “Compiling <module>“ output isn’t meant to be parseable.